api/redisfailover/v1alpha2/types.go

```diff
@@ -39,12 +39,14 @@ type RedisSettings struct {
 	Resources RedisFailoverResources `json:"resources,omitempty"`
 	Exporter  bool                   `json:"exporter,omitempty"`
 	Version   string                 `json:"version,omitempty"`
+	ConfigMap string                 `json:"configMap"`
```

Comment:
Have you think about using an [ObjectReference](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L4293) instead of a string?

Doing this, a ConfigMap from another Namespace could be used, allowing widely used configuration among all the redis-failovers deployed (if wanted).

Reply:
This makes sense. I will try to change it that way.

Reply:
I checked [ConfigMapVolumeSource](https://github.com/kubernetes/api/blob/d216743eed4c3242b85d094d2a589f41a793652d/core/v1/types.go#L1419) which behavior like here. It uses only LocalObjectReference to let the volume use ConfigMap from same namespace. I don't know if it is necessary to let redis use cross namespace ConfigMap? Would this cause any security risk?

Reply:
It's just an idea.

That option would allow using the same configuration for all the redis-failovers deployed.

For example, it you want, by default, having some limits set on every redis, this would allow you to only have one configuration file to maintain.

Both options are perfectly valid. This is just a matter of design and possibilities.